### PR TITLE
Compile translations in the build step

### PR DESCRIPTION
**Pull request by pitaj**

### CODE REVIEW

File: src/plugins.js [hunk 1]
```diff
@@ -29,8 +29,6 @@ var middleware;
 	Plugins.lessFiles = [];
 	Plugins.clientScripts = [];
 	Plugins.acpScripts = [];
-	Plugins.customLanguages = {};
-	Plugins.customLanguageFallbacks = {};
```
Comment (julianlam) [inline src/plugins.js]: What happens to fallbacks now? If a plugin defines `en-GB/foobar.json`, what does accessing `/assets/language/ru/foobar.json` return?

Comment (pitaj) [inline src/plugins.js]: Plugin language fallbacks now work, and apply to every language.

File: src/meta/languages.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
+'use strict';
```
Comment (julianlam) [inline src/meta/languages.js]: Can you add comments to this file? Things like

`// Generate fallback language strings for plugin defined language files`?

File: src/meta/languages.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+module.exports = {
+	build: function buildLanguages(callback) {
+		async.waterfall([
+			getTranslationTree,
+			writeLanguageFiles,
+		], function (err) {
+			if (err) {
+				winston.error('[build] Language build failed');
```
Comment (julianlam) [inline src/meta/languages.js]: `: ' + err.message);` please 😄

File: src/meta/languages.js (hunk 2 above)
Comment (pitaj) [inline src/meta/languages.js]: Done ✔️

File: Gruntfile.js [hunk 4]
```diff
@@ -93,7 +95,15 @@ module.exports = function (grunt) {
 					'!node_modules/nodebb-*/node_modules/**',
 					'!node_modules/nodebb-*/.git/**'
 				]
-			}
+			},
+			langUpdated: {
+				files: [
+					'public/language/**/*.json',
+					'node_modules/nodebb-*/**/*.json',
```
Comment (julianlam) [inline Gruntfile.js]: I see that it's difficult to predict what the language folder would be here... can you add exclusions for `package.json` and `plugin.json`?

**Comment by julianlam** [comment]
Nearly there!! 😄

**Comment by pitaj** [comment]
@julianlam okay I added the change

File: Gruntfile.js (hunk 4 above)
Comment (voanhcuoc) [reply Gruntfile.js]: Hi there @julianlam , I think `plugin.json` should be watched for changes, as developer tend to debug it.

Comment (julianlam) [reply Gruntfile.js]: plugin.json is version controlled and should never be changed in our recommended setup. Any other setups that require this (e.g. Heroku) are outside of the norm and we shouldn't be changing behaviour to suit it.

Comment (voanhcuoc) [reply Gruntfile.js]: But I still have to try and try to config it right in the first place and refactor, add amd modules in the future. And I don't see watching for `plugin.json` will cost anything but deleting one line of code.

**Pull request merged**